include/swift/AST/NameLookupRequests.h

```diff
@@ -302,6 +307,27 @@ class ExpandASTScopeRequest
   void cacheResult(ast_scope::ASTScopeImpl *) const {}
 };
 
+/// Performs unqualified lookup for a DeclName from a given context.
+class UnqualifiedLookupRequest
+    : public SimpleRequest<UnqualifiedLookupRequest,
+                           LookupResult(DeclName, DeclContext *, SourceLoc,
+                                        UnqualifiedLookupFlags),
+                           CacheKind::Uncached> {
+public:
+  using SimpleRequest::SimpleRequest;
+
+private:
+  friend SimpleRequest;
+
+  // Evaluation.
+  // FIXME: We should be taking an OptionSet instead of the raw
+  // UnqualifiedLookupFlags, but we don't want to define == for OptionSet. We
+  // should find a way to define custom equality specifically for requests.
+  llvm::Expected<LookupResult> evaluate(Evaluator &evaluator, DeclName name,
```

Comment:
When faced with these kinds of awful restrictions, I've found that defining a custom type helps.  You can roll back the OptionSet changes if you roll up the parameters to this request into a `LookupRequestDescriptor` or somesuch thing.

Reply:
@CodaFi Good idea! This will also allow us to regain the default parameters.